### PR TITLE
add a favorite button sample

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

File: backend/favorite.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+func setFavorite(w http.ResponseWriter, r *http.Request) {
+	expireInOneDay := time.Now().AddDate(0, 0, 1)
+	favorite := !readFavoriteFromCookie(r)
+	cookie := &http.Cookie{
+		Name:    AMP_FAVORITE_COOKIE,
+		Expires: expireInOneDay,
+		Path:    "/",
+		Value:   strconv.FormatBool(favorite),
+		MaxAge:  36000,
```
Comment (kul3r4) [inline backend/favorite.go]: move 36000 to a variable as we do in request.go

File: src/30_Advanced/Favorite_Button.html [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        cursor: pointer;
+      }
+      </style>
+  </head>
+  <body>
+
+    <!-- 
+    ## Managing State
+
+    We initialize the button state from a [JSON endpoint using amp-state. As we use cookies to identify the user, we need to add the `credentials="include"`  attribute. -->
```
Comment (kul3r4) [inline src/30_Advanced/Favorite_Button.html]: remove [ from [JSON

File: templates/example.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
       {{> css/code.css}}
       {{> css/ampstart.css}}
       {{> css/shared.css }}
-      {{> css/example.css}}
+       {{> css/example.css}}
```
Comment (kul3r4) [inline templates/example.html]: revert this?

File: src/30_Advanced/Favorite_Button.html [hunk 4]
```diff
@@ -0,0 +1,133 @@
+<!--
+  ## Introduction
+
+  This sample demonstrates how to implement a favorite/like/bookmark button in AMP. Our implementation features:
+  
+  * Shows the correct icon based on whether the user already liked an item or not. This works if the AMP is saved
```
Comment (kul3r4) [inline src/30_Advanced/Favorite_Button.html]: if the AMP is saved  -> if the AMP is served

Comment (sebastianbenz) [reply src/30_Advanced/Favorite_Button.html]: done

File: src/30_Advanced/Favorite_Button.html (hunk 2 above)
Comment (sebastianbenz) [reply src/30_Advanced/Favorite_Button.html]: done

File: templates/example.html (hunk 3 above)
Comment (sebastianbenz) [reply templates/example.html]: done

File: backend/favorite.go [hunk 5]
```diff
@@ -41,31 +43,62 @@ func handleFavorite(w http.ResponseWriter, r *http.Request) {
...
+	response := fmt.Sprintf("{ \"value\": %t, \"count\": 123}", favorite)
+	w.Write([]byte(response))
+}
+
+func writeFavorite(w http.ResponseWriter, favorite bool) {
+	response := fmt.Sprintf("%t", favorite)
+	w.Write([]byte(response))
+}
+
+func writeFavoriteCookie(w http.ResponseWriter, r *http.Request, name string, value bool) {
```
Comment (kul3r4) [inline backend/favorite.go]: on top of my head: this file is called favorite.go and yet every method name has "favorite". For example this `writeFavoriteCookie`, couldn't just be writeCookie?

File: src/30_Advanced/Favorite_Button.html [hunk 6]
```diff
@@ -35,21 +35,35 @@
         cursor: pointer;
         border: none;
         margin: 4px;
+        transition: background 300ms ease-in-out;
       }
       amp-list {
         margin: 1rem;
```
Comment (kul3r4) [inline src/30_Advanced/Favorite_Button.html]: Can you use `class=m1`

**Comment by kul3r4** [comment]
@sebastianbenz Added few more comments and please run the lint as the build is failing.

File: backend/favorite.go (hunk 1 above)
Comment (sebastianbenz) [reply backend/favorite.go]: removed

File: backend/favorite.go (hunk 5 above)
Comment (sebastianbenz) [reply backend/favorite.go]: These methods are visible across all go files -> writeCookie is too generic

**Pull request merged**